mapper/orm/src/main/java/org/hibernate/search/mapper/orm/loading/impl/ConditionalExpressionQueryFactory.java

```diff
@@ -35,22 +35,22 @@ public final boolean uniquePropertyIsTheEntityId() {
 	}
 
 	@Override
-	public Query<Long> createQueryForCount(SharedSessionContractImplementor session, EntityMappingType entityMappingType,
+	public Query<Long> createQueryForCount(SharedSessionContractImplementor session, EntityDomainType<?> entityDomainType,
```

Comment:
I'm surprised tests are passing... I'd expect you need to change more than this, since there are other places where we insert entity names in HQL queries.

There's `HqlTypeQueryFactory` in particular, but there may be other places.

Reply:
yeah, I wasn't sure if that's enough, but `HqlTypeQueryFactory` is only used for dynamic entities so the names there are the same, no? 
btw, locally I've updated the ORM helper to always enable the JPA compliance and there were only a couple of tests that failed

Reply:
I double-checked these dynamic map entities, and from what I see, the JPA and ORM names are the same for them and are the ones defined in the XML. I've updated a few existing tests for dynamic entities to be parameterized and test the compliance, so if something changes, we'll notice. 

Thanks for taking a look at this one!